git-server/src/lib.rs

```diff
@@ -403,7 +394,42 @@ async fn git_handler(
         }
     }
 
-    Ok::<_, Error>((status, response_headers, body))
+    Ok((status, response_headers, body).into_response())
+}
+
+async fn git_handler(
+    Extension(ctx): Extension<Context>,
+    AxumPath((project_id, request)): AxumPath<(String, String)>,
+    method: Method,
+    headers: HeaderMap,
+    body: Bytes,
+    ConnectInfo(remote): ConnectInfo<SocketAddr>,
+    query: RawQuery,
+) -> Response {
+    let peer_id = None;
+    let query = query.0.unwrap_or_default();
+
+    let urn = match resolve_project_id(&ctx, &project_id).await {
+        Ok(urn) => urn,
+        Err(Error::AliasNotFound) | Err(Error::InvalidId) => {
+            return (StatusCode::NOT_FOUND, format!("Not found: {}", project_id)).into_response()
+        }
+        Err(e) => return e.into_response(),
+    };
+
+    let (status, headers, body) = match git(
+        ctx, method, headers, body, remote, urn, peer_id, &request, query,
+    )
+    .await
+    {
+        Ok((status, headers, body)) => (status, headers, body),
+        Err(e) => return e.into_response(),
+    };
+
+    match axum_resp_from_status_headers_body_triple(status, headers, body) {
+        Ok(resp) => resp,
+        Err(e) => e.into_response(),
+    }
```

Comment:
What was the reason to move this out?

Reply:
Every occurrence of `?` would otherwise have to be replaced with `match […] { …, Err(e) => return e.into_response() }`.  If it’s factored out, it’s still possible to use `?` and handle all the errors in a single place.